kong/clustering.lua

```diff
@@ -369,11 +376,37 @@ local function push_config_timer(premature, semaphore, delay)
   end
 
   while not exiting() do
-    local ok, err = semaphore:wait(10)
+    local ok, err = semaphore:wait(1)
+    if exiting() then
+      return
+    end
     if ok then
       ok, err = pcall(push_config)
       if ok then
-        ngx.sleep(delay)
+        if delay <= 1 then
```

Comment:
I think this first `if` might be unnecessary – If I understand the code correctly, the other branch of the `if` will also call `ngx.sleep(delay)` in the first iteration of the `while` loop (plus it will check `exiting()`, which I think is a good thing here?). So I recommend replacing the `if` with the part in the `else`